pkg/reconciler/pingsource/pingsource.go

```diff
@@ -178,7 +178,7 @@ func needsUpdating(ctx context.Context, oldDeploymentSpec *appsv1.DeploymentSpec
 		return false, nil
 	}
 
-	return zero(oldDeploymentSpec.Replicas) || !equality.Semantic.DeepEqual(container.Env, newEnvVars), container
+	return zero(oldDeploymentSpec.Replicas) || !equality.Semantic.DeepDerivative(newEnvVars, container.Env), container
```

Comment:
DeepDerivative: we ignore unset fields, ok - why the change of the arguments?

Reply:
DeepDerivative: we ignore unset fields of the first argument, so we need set "expected" as the first argument and current as the second one